src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -22,20 +21,54 @@ export interface IDownloadFileRow {
 }
 
 export function generateOqlData(datum: GeneticTrackDatum,
-                                geneAlterationDataByGene?: {[gene: string]: IGeneAlteration}): IOqlData
+                                geneAlterationDataByGene?: {[gene: string]: IGeneAlteration},
+                                molecularProfileIdToMolecularProfile?: {[molecularProfileId:string]:MolecularProfile}): IOqlData
 {
     const proteinChanges: string[] = [];
     const fusions: string[] = [];
     const cnaAlterations: ISubAlteration[] = [];
     const proteinLevels: ISubAlteration[] = [];
     const mrnaExpressions: ISubAlteration[] = [];
-
+    const alterationTypes: string[] = [];
+    let isMutationNotProfiled = false;
+    let isFusionNotProfiled = false;
+    let isCnaNotProfiled = false;
+    let isMrnaExpNotProfiled = false;
+    let isProteinLevelNotProfiled = false;
+
+    //record the profile information
+    if (datum.not_profiled_in)
+    {
+        for (const profile of datum.not_profiled_in)
+        {
+            if (molecularProfileIdToMolecularProfile)
+            {
+                const molecularAlterationType = molecularProfileIdToMolecularProfile[profile.molecularProfileId].molecularAlterationType;
+                switch (molecularAlterationType)
+                {
+                    case AlterationTypeConstants.COPY_NUMBER_ALTERATION:
+                        isCnaNotProfiled = true;
+                        break;
+                    case AlterationTypeConstants.MRNA_EXPRESSION:
+                        isMrnaExpNotProfiled = true;
+                        break;
+                    case AlterationTypeConstants.PROTEIN_LEVEL:
+                        isProteinLevelNotProfiled = true;
+                        break;
+                    case AlterationTypeConstants.MUTATION_EXTENDED:
+                        isMutationNotProfiled = true;
+                    case AlterationTypeConstants.FUSION:
+                        isFusionNotProfiled = true;
+                        break;
```

Comment:
would be much simpler to store this data as a map `const isNotProfiled ={[alterationTypeConstant:string]:boolean}`. Then lines 47-63 can be compressed to `isNotProfiled[molecularAlterationType] = true`